apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ConfigsImportService.java

```diff
@@ -16,122 +16,444 @@
-   * @throws AccessControlException if has no modify namespace permission
-   */
-  public void importOneConfigFromFile(
-      final String env,
-      final String standardFilename,
-      final InputStream inputStream
-  ) {
-    final String configText;
-    try(InputStream in = inputStream) {
-      configText = ConfigToFileUtils.fileToString(in);
+  private void importNamespace(final String appId, final Env env,
+                               final String clusterName, final String namespaceName,
+                               final String configText, final String format,
+                               boolean ignoreConflictNamespace) {
+    NamespaceDTO namespaceDTO;
+    try {
+      namespaceDTO = namespaceService.loadNamespaceBaseInfo(appId, env, clusterName, namespaceName);
+    } catch (Exception e) {
+      //not existed
+      namespaceDTO = null;
+    }
+
+    if (namespaceDTO == null) {
+      namespaceDTO = new NamespaceDTO();
+      namespaceDTO.setAppId(appId);
+      namespaceDTO.setClusterName(clusterName);
+      namespaceDTO.setNamespaceName(namespaceName);
+      namespaceDTO.setDataChangeCreatedBy(currentUser);
+      namespaceDTO.setDataChangeLastModifiedBy(currentUser);
+      namespaceDTO = namespaceService.createNamespace(env, namespaceDTO);
+    }
+
+    List<ItemDTO> itemDTOS = itemService.findItems(appId, env, clusterName, namespaceName);
+    // skip import if target namespace has existed items
+    if (!CollectionUtils.isEmpty(itemDTOS) && ignoreConflictNamespace) {
+      return;
+    }
+
+    this.importItems(appId, env, clusterName, namespaceName, namespaceDTO.getId(), format, configText);
+  }
+
+  private void importItems(final String appId, final Env env, final String clusterName,
+                           final String namespaceName, final long namespaceId, final String format,
+                           final String configText) {
+    final NamespaceTextModel model = new NamespaceTextModel();
+
+    model.setAppId(appId);
+    model.setEnv(env.getName());
+    model.setClusterName(clusterName);
+    model.setNamespaceName(namespaceName);
+    model.setNamespaceId(namespaceId);
+    model.setFormat(format);
+    model.setConfigText(configText);
+    model.setOperator(currentUser);
+
+    itemService.updateConfigItemByText(model);
+  }
+
+  private String readContent(ZipInputStream zipInputStream) {
+    try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+      byte[] buffer = new byte[1024];
+      int offset;
+      while ((offset = zipInputStream.read(buffer)) != -1) {
+        out.write(buffer, 0, offset);
+      }
+      return out.toString("UTF-8");
     } catch (IOException e) {
-      throw new ServiceException("Read config file errors:{}", e);
+      LOGGER.error("Read file content from zip error.", e);
+      return null;
+    }
+  }
+
+  static class NamespaceImportData {
```

Comment:
Please implement the toString method so that it could be printed friendly in line 239

Reply:
done